File: prog1.cpp
//two sum
#include <iostream>
#include <vector>
#include <unordered_map>
using namespace std;

vector<int> twoSum(vector<int> numbers, int target)
{
    unordered_map<int,int> m;
    vector<int> ans;
    for (int i=0;i<numbers.size();i++)
    {
        if (m.find(numbers[i])==m.end())
        {
            m[target-numbers[i]]=i;
        }
        else
        {
            ans.push_back(m[numbers[i]]+1);
            ans.push_back(i+1);
            break;
        }
    }
    return ans;
}


int main()
{
    vector<int> numbers={2,7,11,15};
    int target=26;
    vector<int> ans=twoSum(numbers, target);
    cout << ans[0] << " " << ans[1] << endl;
    return 0;
}

File: prog1.py
def twoSum(numbers, target):
    d={}
    for i in range(len(numbers)):
        if numbers[i] not in d:
            d[target-numbers[i]] = i
            print i
            print d
        else:
            break
    
    return d[numbers[i]]+1, i+1
    #return i

numbers=[2,7,11,15]
target=18
print twoSum(numbers,target)

